Bottom: 897cee0b17f17050fe3667cbaf7c6d9d3f66ab9d
Top:    24c419a565e8be47500329bc1ba94c160d8d19e1
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-08-22 15:38:45 -0700

Refresh of dbg-fork

---

diff --git a/src/preload.c b/src/preload.c
index d1d8b06..d0fead2 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -570,11 +570,13 @@ static void fork_passive(int socket)
 
 	len = sizeof sin6;
 	ret = real.getsockname(sfd, (struct sockaddr *) &sin6, &len);
+	printf("fork_passive - getsockname %d\n", ret);
 	if (ret)
 		goto out;
 	sin6.sin6_flowinfo = sin6.sin6_scope_id = 0;
 	memset(&sin6.sin6_addr, 0, sizeof sin6.sin6_addr);
 
+	sem_unlink("/rsocket_fork");
 	sem = sem_open("/rsocket_fork", O_CREAT | O_RDWR,
 		       S_IRWXU | S_IRWXG, 1);
 	if (sem == SEM_FAILED) {
